### PR TITLE
Update index.d.ts with optionals

### PR DESCRIPTION
**Pull request by stevarino**
Trying to learn some typescript and this made the red squigglies go away.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -149,37 +149,36 @@ declare module "bedrock-protocol" {
...
+    logging?: boolean,
     // Skip authentication for connecting clients?
-    offline: false,
+    offline?: false,
     // Specifies which game edition to sign in as to the destination server. Optional, but some servers verify this.
-    authTitle: title | string
+    authTitle?: title | string
     // Where to proxy requests to.
-    destination: {
+    destination?: {
```
Comment (extremeheat) [inline index.d.ts]: this field should be required

Comment (stevarino) [reply index.d.ts]: Fixed, thanks! :-)

**Review by extremeheat — APPROVED** [review]

**Review by hvlxh — APPROVED** [review]

**Comment by extremeheat** [comment]
Thanks!

**Comment by extremeheat** [comment]
Thanks!

**Pull request merged**